src/web/gwc/src/main/java/org/geoserver/gwc/web/GWCIconFactory.java

```diff
@@ -84,6 +86,16 @@ public static CachedLayerType getCachedLayerType(final TileLayer layer) {
         if (layer instanceof WMSLayer) {
             return CachedLayerType.GWC;
         }
+
+        List<GWCTileLayerIconCustomizer> iconCustomizers =
+                GeoServerExtensions.extensions(GWCTileLayerIconCustomizer.class);
+        for (GWCTileLayerIconCustomizer iconCustomizer : iconCustomizers) {
+            CachedLayerType cachedLayerType = iconCustomizer.getCachedLayerType(layer);
+            if (cachedLayerType != null) {
```

Comment:
So here it checks for null, but the implementation uses the "unkown" constant instead.
Think of an expected behavior in the interface that would allow multiple customizers to exist at the same time.

Reply:
yep, I have updated both the javadoc of the interface as well as these checks to keep scanning when the return is UNKNOWN.